eclair-core/src/test/scala/fr/acinq/eclair/payment/send/OfferPaymentSpec.scala

```diff
@@ -0,0 +1,134 @@
+import fr.acinq.eclair.wire.protocol.OfferTypes.{InvoiceRequest, Offer, PaymentInfo}
+import fr.acinq.eclair.wire.protocol.{OnionMessagePayloadTlv, TlvStream}
+import fr.acinq.eclair.{CltvExpiryDelta, Features, MilliSatoshiLong, NodeParams, TestConstants, randomBytes32, randomKey}
+import org.scalatest.Outcome
+import org.scalatest.funsuite.FixtureAnyFunSuiteLike
+import scodec.bits.HexStringSyntax
+
+import java.util.UUID
+import scala.concurrent.duration.DurationInt
+
+class OfferPaymentSpec extends ScalaTestWithActorTestKit(ConfigFactory.load("application")) with FixtureAnyFunSuiteLike {
+
+  case class FixtureParam(offerPayment: ActorRef[Command], nodeParams: NodeParams, postman: TypedProbe[Postman.Command], paymentInitiator: akka.testkit.TestProbe, routeParams: RouteParams)
+
+  override def withFixture(test: OneArgTest): Outcome = {
+    val nodeParams = TestConstants.Alice.nodeParams
+    val postman = TypedProbe[Postman.Command]("postman")
+    val paymentInitiator = akka.testkit.TestProbe("paymentInitiator")(system.toClassic)
+    val offerPayment = testKit.spawn(OfferPayment(nodeParams, postman.ref, paymentInitiator.ref))
+    val routeParams = nodeParams.routerConf.pathFindingExperimentConf.getRandomConf().getDefaultRouteParams
+    try {
+      withFixture(test.toNoArgTest(FixtureParam(offerPayment, nodeParams, postman, paymentInitiator, routeParams)))
+    } finally {
+      testKit.stop(offerPayment)
+    }
+  }
+
+  implicit val classicSystem: ActorSystem = system.classicSystem
+
+  test("basic offer payment") { f =>
+    import f._
+
+    val probe = TestProbe()
+    val merchantKey = randomKey()
+
+    val offer = Offer(None, "amountless offer", merchantKey.publicKey, Features.empty, nodeParams.chainHash)
+    offerPayment ! PayOffer(probe.ref, offer, 40_000_000 msat, 1, SendPaymentConfig(None, 1, routeParams, blocking = false))
+    val Postman.SendMessage(_, Recipient(recipientId, _, _), _, message, replyTo, _) = postman.expectMessageType[Postman.SendMessage]
+    assert(recipientId == merchantKey.publicKey)
+    assert(message.get[OnionMessagePayloadTlv.InvoiceRequest].nonEmpty)
+    val Right(invoiceRequest) = InvoiceRequest.validate(message.get[OnionMessagePayloadTlv.InvoiceRequest].get.tlvs)
+
+    val preimage = randomBytes32()
+    val paymentRoute = PaymentBlindedRoute(RouteBlinding.create(randomKey(), Seq(merchantKey.publicKey), Seq(hex"7777")).route, PaymentInfo(0 msat, 0, CltvExpiryDelta(0), 0 msat, 1_000_000_000 msat, Features.empty))
+    val invoice = Bolt12Invoice(invoiceRequest, preimage, merchantKey, 1 minute, Features.empty, Seq(paymentRoute))
+    replyTo ! Postman.Response(FinalPayload(TlvStream(OnionMessagePayloadTlv.Invoice(invoice.records)), TlvStream.empty))
+    val send = paymentInitiator.expectMsgType[SendPaymentToNode]
+    assert(send.invoice == invoice)
+
+    val paymentId = UUID.randomUUID()
+    send.replyTo ! paymentId
+    val uuid = probe.expectMsgType[UUID]
+    assert(uuid == paymentId)
```

Comment:
This isn't testing anything is it? You're both sending the message and verifying you receive it inside the test...I believe you can just remove these 4 lines.